src/Grpc.Net.Client/Balancer/Subchannel.cs

```diff
@@ -327,6 +327,15 @@ private async Task ConnectTransportAsync()
                 return;
             }
 
+            // There is already a connect in-progress on this transport.
+            // Don't cancel and start again as that causes queued requests waiting on the connect to fail.
+            if (_connectContext != null && !_connectContext.Disposed)
```

Comment:
I had to check the file to see if this had a race condition (i.e. can `_connectContext` *become* `null`; it can't) - it *isn't*, but may I suggest:

``` c#
if (_connectContext is { Disposed: false })
```

so that it is obviously not a race (i.e. single-read)?